### PR TITLE
Added implementation of CSRF validation to POST controller action.

### PR DESCRIPTION
**Pull request by dudzio12**
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/routing.html#action-class

### CODE REVIEW

**Comment by dudzio12** [comment]
That's for Magento 2.3 compatibility.
All requests from PAYU were declined and end with 302 code.

File: Controller/Data/GetNotify.php [hunk 1]
```diff
@@ -57,6 +60,30 @@ public function __construct(
...
 
+    /**
+     * Create exception in case CSRF validation failed.
+     * Return null if default exception will suffice.
+     *
+     * @param RequestInterface $request
+     *
+     * @return InvalidRequestException|null
+     */
+    public function createCsrfValidationException(RequestInterface $request): ?InvalidRequestException {
```
Comment (regdos) [inline Controller/Data/GetNotify.php]: Please remove ": ?InvalidRequestException". It isn't compatible with old php.

File: Controller/Data/GetNotify.php [hunk 2]
```diff
@@ -57,6 +60,30 @@ public function __construct(
...
+
+    /**
+     * Perform custom request validation.
+     * Return null if default validation is needed.
+     *
+     * @param RequestInterface $request
+     *
+     * @return bool|null
+     */
+    public function validateForCsrf(RequestInterface $request): ?bool {
```
Comment (regdos) [inline Controller/Data/GetNotify.php]: Please remove ": ?bool". It isn't compatible with old php.

**Comment by dudzio12** [comment]
Done

**Comment by dudzio12** [comment]
I've reverted those changes, due to Magento 2.3 official requirements:
`PHP 7.1.3+, 7.2.x
`So there is no support for PHP <7.1.

It also broke compatibility with `CsrfAwareActionInterface` interface, so it must be like this.

**Comment by regdos** [comment]
We added new repo for Magento 2.3 https://github.com/PayU/plugin_magento_23 with this change.

Thank's for You help.

**Pull request closed without merging**